pkg/cmd/generate-pg-catalog/main.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+// This connects to Postgresql and retrieves all the information about
+// the tables in the pg_catalog schema. The result is printed into a
+// comma separated lines which are meant to be store at a CSV and used
+// to test that cockroachdb pg_catalog is up to date.
+//
+// This accepts the following arguments:
+//
+// -user: to change default pg username of `postgres`
+// -addr: to change default pg address of `localhost:5432`
+//
+// Output of this file should generate:
+// pkg/sql/testdata/pg_catalog_tables
+package main
+
+import (
+	"flag"
+	"fmt"
+
+	"github.com/jackc/pgx"
+)
+
+// This query uses udt_name::regtype instead of data_type column because
+// data_type only says "ARRAY" but does not say which kind of array it is.
+const getPgCatalogSQL = `
+	SELECT table_name, column_name, udt_name::regtype data_type, udt_name::regtype::oid data_type_oid
```

Comment:
i've been researching this query and i think we actually cannot use it. it isn't safe to use because the cast from name to oid (`udt_name::regtype::oid`) loses information. the query should instead be like this:

```
SELECT
  c.relname AS table_name,
  a.attname AS column_name,
  t.typname AS data_type,
  t.oid AS data_type_oid
FROM pg_class c
JOIN pg_attribute a ON a.attrelid = c.oid
JOIN pg_type t ON t.oid = a.atttypid
JOIN pg_namespace n ON n.oid = c.relnamespace
WHERE n.nspname = 'pg_catalog'
ORDER BY 1, 2;
```

Reply:
Done